proposals/20230530-driver-kernel-testing-framework.md

```diff
@@ -0,0 +1,454 @@
+
+*Compiler Versions*
+
+Select the most appropriate compiler version and build container for the CI-integrated tests. Apart from the compiler version, the GLIBC version in the build container can also have an impact on the ability to compile the driver for a given kernel.
+
+> The expanded CI tests may necessitate the use of approximately 30 low-resource virtual machines (VMs) that can run continuously 24/7. These VMs would be distributed across multiple third-party cloud providers. Alternatively, VMs with KVM on hosted GitHub runners can be utilized, or an equivalent suitable solution can be adopted. To adequately cover the condensed kernel test grid, it is estimated that up to 70 test runs would be required for each testing cycle. These tests can be launched using GitHub workflows leveraging SSH remote commands. The test results are then retrieved through this method as well. Initially, it would be logical to support these tests on demand only (such as nightly tests) to avoid simultaneous runs that may try to access the same VM at the same time. In addition to the test VMs, it may be necessary to expand the CI workflows in terms of builder containers.
+
+Please refer to Appendix 1 for a concrete example of a possible kernel test grid.
+
+
+## Outlook
+
+The following possibilities serve as an outlook for future enhancements. These potential improvements are anticipated after the release of Falco 0.36.
+
+- Objective: Conduct comprehensive regression testing, cost budgeting, and benchmarking.
+- Actions: Allocate additional resources and budget to expand the testing infrastructure, enabling thorough regression testing and performance evaluation with realistic workloads and simulations of production settings. Furthermore, it is suggested that the kernel test grid be made more dynamic.
+- Expected Outcome: Improved detection of regressions, optimized cost budgeting for kernel monitoring, and benchmarking for performance optimization.
+
+The successful implementation depends on increased CI budgeting and, more importantly, the availability of additional engineering resources. Additionally, we are actively collaborating with the CNCF TAG Environmental Sustainability to establish core indices that reflect the cost implications of kernel monitoring for threat detection. This collaboration ensures that we adopt a sustainable and compatible approach, taking into account various cost factors. You can find more information on our activities and best practices for CNCF projects in the [Proof of Environmental Sustainability activities and best practices for CNCF projects](https://github.com/cncf/tag-env-sustainability/issues/64#issuecomment-1496197590) issue.
+
+
+## Resources
+
+- CI [Github Workflows](https://github.com/falcosecurity/libs/tree/master/.github/workflows)
+- CI [CircleCI](https://github.com/falcosecurity/libs/tree/master/.circleci) 
```

Comment:
```suggestion
```

We want to abandon [CircleCI in favor of GitHub Action](https://github.com/falcosecurity/falco/issues/2454)